plugins/repository-s3/src/main/java/org/elasticsearch/plugin/repository/s3/S3RepositoryPlugin.java

```diff
@@ -144,6 +145,7 @@ public void onModule(SettingsModule settingsModule) {
         settingsModule.registerSetting(S3Repository.Repository.STORAGE_CLASS_SETTING);
         settingsModule.registerSetting(S3Repository.Repository.CANNED_ACL_SETTING);
         settingsModule.registerSetting(S3Repository.Repository.BASE_PATH_SETTING);
+        settingsModule.registerSetting(S3Repository.Repository.THROTTLE_RETRIES_SETTING);
```

Comment:
> I totally agree but this should be in a follow up PR.

Would love to have another solution in this PR :/

If we can't make it, then can you please add a todo/comment line here and create an issue? Thanks